targets/TARGET_Cypress/TARGET_PSOC6/cy_us_ticker_api.c

```diff
@@ -54,6 +54,23 @@ static void cy_us_ticker_irq_handler(MBED_UNUSED void *arg, MBED_UNUSED cyhal_ti
 void us_ticker_init(void)
 {
     if (!cy_us_ticker_initialized) {
+
+#ifdef TARGET_TFM
```

Comment:
This does not make sense to me. If these are only used for testing, why aren't they being reserved by the test itself. That keeps the code together and avoid an unnecessary reservation in the general code.  Also, it says it uses channels 1 & 2, but then reserves 0 & 1.

Reply:
My understanding here: us ticker is trying to allocate 2 free timers. Alocate 0 - failed, try next one, failed, next one - until there is a free one. Considering all channels are capable of what us ticker requires.
`TARGET_TFM` should not be necessary here, as any app can allocate timer before this runs. 

@morser499 What would you suggest here ?

Reply:
@0xc0170 How can an app allocate a timer before this runs?

Reply:
Unfortunatly it looks like I posted twice...
@0xc0170 cyhal_timer_init() will already allocate the first available one. The calls to cyhal_hwmgr_reserve() just prevent those resources from being used.
@Patater I am not sure how the application is actually using these. Can you point me to the application in question to review?

Reply:
It uses the hardware directly, not using Mbed OS APIs. It's a bare metal application that we are avoiding modifying for maintainability reasons: it's best if we can avoid any modifications to the application that aren't upstream.

This is the file the application uses for running the timer. The application assumes there is some timer available to NS, available via a HAL-like function. https://git.trustedfirmware.org/trusted-firmware-m.git/tree/platform/ext/target/cypress/psoc64/plat_test.c#n55

Reply:
I understand not wanting to change a code example, but putting this code here increases maintainability overhead. The code would be spread out in two completely different places.